actors/builtin/miner/miner_test.go

```diff
@@ -1114,146 +1165,98 @@ func TestProvingPeriodCron(t *testing.T) {
 		WithBalance(bigBalance, big.Zero())
 
 	t.Run("empty periods", func(t *testing.T) {
-		t.Skip("Disabled in miner state refactor #648, restore soon")
 		rt := builder.Build(t)
 		actor.constructAndVerify(rt)
 		st := getState(rt)
 		assert.Equal(t, periodOffset, st.ProvingPeriodStart)
 
-		// First cron invocation just before the first proving period starts.
-		rt.SetEpoch(periodOffset - 1)
+		// crons before proving period do nothing
 		secondCronEpoch := periodOffset + miner.WPoStProvingPeriod - 1
-		actor.onDeadlineCron(rt, &cronConfig{
-			expectedEntrollment: secondCronEpoch,
-		})
+		dlinfo := actor.deadline(rt)
+		for dlinfo.Close < secondCronEpoch {
+			dlinfo = advanceDeadline(rt, actor, &cronConfig{})
+		}
+
 		// The proving period start isn't changed, because the period hadn't started yet.
 		st = getState(rt)
 		assert.Equal(t, periodOffset, st.ProvingPeriodStart)
 
-		rt.SetEpoch(secondCronEpoch)
-		actor.onDeadlineCron(rt, &cronConfig{
-			expectedEntrollment: periodOffset + 2*miner.WPoStProvingPeriod - 1,
```

Comment:
hmm enrollment still happens right?  If so it looks like we are no longer testing that it does.

Reply:
Enrollment still happens. The expectation is in `advanceDeadline`, so it is under test. The fact that you called it out, suggests that it should be more visible in the test. I'll use the more explicit version.